packer/conf/buildkite-agent/hooks/post-command

```diff
@@ -1,3 +1,27 @@
 #!/bin/bash
 
-ssh-agent -k
\ No newline at end of file
+set -eux -o pipefail
+
+ssh-agent -k
+
+if [[ -n "${BUILDKITE_DOCKER_COMPOSE_CONTAINER:-}" && -n ${DOCKER_CACHE_IMAGES_PUSH:-} && BUILDKITE_COMMAND_EXIT_STATUS -eq 0 ]] ; then
+    echo "~~~ Pushing cached docker images"
+    (
+        # clear any aws env vars so the machine profile is used
+        unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_REGION
+
+        slug=/tmp/docker-cache/$BUILDKITE_PIPELINE_SLUG.tar.gz
+        images_file=s3://$BUILDKITE_IMAGE_CACHE_BUCKET/$BUILDKITE_PIPELINE_SLUG.images
+        images=$(echo $(docker images -a | grep $(echo $BUILDKITE_JOB_ID | sed 's/-//g') | awk '{print $1}' | xargs -n 1 docker history -q | grep -v '<missing>'))
+
+        if [[ -n $images ]] && ( ! aws s3 ls $images_file || [[ "$images" != $(aws s3 cp $images_file -) ]]) ; then
+            rm -rf /tmp/docker-cache
+            mkdir -p /tmp/docker-cache
+
+            docker save $images | gzip -c > $slug
```

Comment:
Just happened to be reading this, `buildcache` doesn't save the whole images, only the configuration json. Not sure if it fits better or worse for you because of it. Also `save $(docker history)` was buggy in docker `v1.11` and buildcache works around that.